django/test/utils.py

```diff
@@ -85,11 +86,7 @@ def keys(self):
         """
         Flattened keys of subcontexts.
         """
-        keys = set()
-        for subcontext in self:
-            for dict in subcontext:
-                keys |= set(dict.keys())
-        return keys
+        return set(chain.from_iterable(dict.keys() for subcontext in self for dict in subcontext))
```

Comment:
Ditto about the `keys()` call. I'd also try to avoid cloaking the `dict` builtin while you are around.

```python
set(chain.from_iterable(dict_ for subcontext in self for dict_ in subcontext))
```